### ISSUE TITLE
Amplify datastore does not sync with dynamodb when an item is added through lambda

### ISSUE THREAD

**Issue by ori9998** [aws-amplify/amplify-js#10537]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React Native

### Amplify APIs

DataStore

### Amplify Categories

api

### Environment information

<details>

```
# Put output below this line


System:
    OS: macOS 12.4
    CPU: (8) arm64 Apple M1
    Memory: 95.88 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 18.4.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.13.1 - /opt/homebrew/bin/npm
    Watchman: 2022.06.13.00 - /opt/homebrew/bin/watchman
  Browsers:
    Chrome: 106.0.5249.119
    Safari: 15.5
  npmPackages:
    @babel/core: ^7.12.9 => 7.19.3 
    @babel/runtime: ^7.12.5 => 7.19.4 
    @notifee/react-native: ^7.0.4 => 7.0.4 
    @react-native-async-storage/async-storage: ^1.17.10 => 1.17.10 
    @react-native-clipboard/clipboard: ^1.11.1 => 1.11.1 
    @react-native-community/checkbox: ^0.5.12 => 0.5.12 
    @react-native-community/eslint-config: ^2.0.0 => 2.0.0 
    @react-native-community/netinfo: ^9.3.5 => 9.3.5 
    @react-native-firebase/app: ^15.7.0 => 15.7.0 
    @react-native-firebase/messaging: ^15.7.0 => 15.7.0 
    @react-native-picker/picker: ^2.4.7 => 2.4.7 
    @react-navigation/bottom-tabs: ^6.4.0 => 6.4.0 
    @react-navigation/material-top-tabs: ^6.2.4 => 6.2.4 
    @react-navigation/native: ^6.0.13 => 6.0.13 
    @react-navigation/native-stack: ^6.9.0 => 6.9.0 
    @types/jest: ^29.1.2 => 29.1.2 
    @types/react: ^18.0.21 => 18.0.21 
    @types/react-native: ^0.70.4 => 0.70.4 
    @types/react-native-vector-icons: ^6.4.12 => 6.4.12 
    @types/react-test-renderer: ^18.0.0 => 18.0.0 
    @types/uuid: ^8.3.4 => 8.3.4 
    HelloWorld:  0.0.1 
    amazon-cognito-identity-js: ^5.2.11 => 5.2.11 
    aws-amplify: ^4.3.39 => 4.3.39 
    aws-amplify-react-native: ^6.0.5 => 6.0.5 
    babel-jest: ^26.6.3 => 26.6.3 
    date-fns: ^2.29.3 => 2.29.3 
    eslint: ^7.32.0 => 7.32.0 
    formik: ^2.2.9 => 2.2.9 
    hermes-inspector-msggen:  1.0.0 
    i18next: ^21.10.0 => 21.10.0 
    jest: ^26.6.3 => 26.6.3 
    metro-react-native-babel-preset: ^0.70.3 => 0.70.3 
    moment: ^2.29.4 => 2.29.4 
    react: 18.0.0 => 18.0.0 
    react-i18next: ^11.18.6 => 11.18.6 
    react-native: 0.69.5 => 0.69.5 
    react-native-get-random-values: ^1.8.0 => 1.8.0 
    react-native-image-crop-picker: ^0.38.0 => 0.38.0 
    react-native-localize: ^2.2.4 => 2.2.4 
    react-native-modal: ^13.0.1 => 13.0.1 
    react-native-pager-view: ^6.0.1 => 6.0.1 
    react-native-safe-area-context: ^4.4.1 => 4.4.1 
    react-native-screens: ^3.18.0 => 3.18.0 
    react-native-svg: ^13.4.0 => 13.4.0 
    react-native-svg-transformer: ^1.0.0 => 1.0.0 
    react-native-tab-view: ^3.2.1 => 3.2.1 
    react-native-vector-icons: ^9.2.0 => 9.2.0 
    react-native-version-check: ^3.4.3 => 3.4.3 
    react-test-renderer: 18.0.0 => 18.0.0 
    typescript: ^4.8.4 => 4.8.4 
    uuid: ^9.0.0 => 9.0.0 (3.4.0, 3.3.2, 8.3.2, 7.0.3)
    yup: ^0.32.11 => 0.32.11 
  npmGlobalPackages:
    @aws-amplify/cli: 10.3.1
    firebase-tools: 11.8.0
    npm: 8.13.1
    react-native-vector-icons: 9.2.0
    rnpm: 1.9.0

</details>


### Describe the bug

I'm having a hard time syncing dynamodb and datastore. 

I'm using lambda (post confimation)to add user to dynamodb when new user creates an account.

However, the problem is that when a new user is added to dynamodb through lambda, my app (react native) cannot query that new user from datastore.

I have read a similar bug : [link](https://github.com/aws-amplify/amplify-js/issues/9842)  but he uses graphql operations within the lambda function, but I just use datastore.

So, to sum up:

1. Is there a way to sync between dynamodb and datastore from the front-end side?
2. Can I use both datastore and graphql (in lambda use graphql and use datastore to query, update within the app)?

### Expected behavior

When an item is added through lambda to dynamodb, I would be able to query that item in the front end side (datastore) instantly (without reloading or signing in and out)

### Reproduction steps

1. create a simple app
2. run the app
3. by using lambda, add an item to dynamodb
4. without reloading or signing in and out, try to query the added item
5. dynamodb and datastore would not sync 

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by sagardesai1** [comment]
what was the resolution here?

**Comment by chrisbonifacio** [comment]
@sagardesai1 to trigger subscriptions on clients using DataStore, a GraphQL mutation must be performed through AppSync so I assume that's what they did. Inserting a record directly into DynamoDB will not trigger AppSync subscriptions.
